src/coreclr/scripts/superpmi_asmdiffs.py

```diff
@@ -209,33 +211,63 @@ def main(main_args):
         "-target_os", platform_name,
         "-target_arch", arch_name,
         "-arch", host_arch_name,
-        "-base_jit_path", base_jit_path,
-        "-diff_jit_path", diff_jit_path,
+        "-base_jit_path", base_checked_jit_path,
+        "-diff_jit_path", diff_checked_jit_path,
         "-spmi_location", spmi_location,
         "-error_limit", "100",
         "-log_level", "debug",
         "-log_file", log_file,
         "-retainOnlyTopFiles"])
 
+    print("Running superpmi.py tpdiff")
+
+    overall_md_tpdiff_summary_file = os.path.join(spmi_location, "tpdiff_summary.md")
+    if os.path.isfile(overall_md_tpdiff_summary_file):
+        os.remove(overall_md_tpdiff_summary_file)
+
+    _, _, return_code = run_command([
+        python_path,
+        os.path.join(script_dir, "superpmi.py"),
+        "tpdiff",
+        "--no_progress",
+        "-core_root", core_root_dir,
+        "-target_os", platform_name,
+        "-target_arch", arch_name,
+        "-arch", host_arch_name,
+        "-base_jit_path", base_release_jit_path,
+        "-diff_jit_path", diff_release_jit_path,
+        "-spmi_location", spmi_location,
+        "-error_limit", "100",
+        "-log_level", "debug",
+        "-log_file", log_file])
+
     # If there are asm diffs, and jit-analyze ran, we'll get a diff_summary.md file in the spmi_location directory.
     # We make sure the file doesn't exist before we run diffs, so we don't need to worry about superpmi.py creating
     # a unique, numbered file. If there are no diffs, we still want to create this file and indicate there were no diffs.
 
-    overall_md_summary_file_target = os.path.join(log_directory, "superpmi_diff_summary_{}_{}.md".format(platform_name, arch_name))
-    if os.path.isfile(overall_md_summary_file):
-        try:
-            print("Copying summary file {} -> {}".format(overall_md_summary_file, overall_md_summary_file_target))
-            shutil.copy2(overall_md_summary_file, overall_md_summary_file_target)
-        except PermissionError as pe_error:
-            print('Ignoring PermissionError: {0}'.format(pe_error))
-    else:
-        # Write a basic summary file. Ideally, we should not generate a summary.md file. However, currently I'm seeing
-        # errors where the Helix work item fails to upload this specified file if it doesn't exist. We should change the
-        # upload to be conditional, or otherwise not error.
-        with open(overall_md_summary_file_target, "a") as f:
-            f.write("""\
-No diffs found
-""")
+    md_files = []
+
+    overall_md_asmdiff_summary_file_target = os.path.join(log_directory, "superpmi_diff_summary_{}_{}.md".format(platform_name, arch_name))
```

Comment:
Regarding the output table, I am only interested in the actual diff rather than raw numbers. I would suggest to just display the diff % (if possible, in tabular form) and then have an expansion that shows the raw numbers along with diffs. That way, anyway who is looking at the extension page can quickly scan through the numbers to know if TP was regressed or not. Also, leave a note about "lower is better" .

Reply:
Good idea, let me see if I can do that in a nice way. It is complicated a little by how we communicate the summaries back (as markdown files instead of just the data).

Reply:
This is how it looks now: https://dev.azure.com/dnceng/public/_build/results?buildId=1731173&view=ms.vss-build-web.run-extensions-tab
What do you think?

Reply:
Much better. Looking at the title "Throughput impact on Windows x64", it makes me think that is this the right way to measure the throughput impact for binaries compiled for Linux platforms? On Linux, we use LLVM which would generate different instructions, while here, we are measuring the count of instructions generated by MSVC on cross-compiled clrjit.

Reply:
Yes, it is somewhat misleading. However I spoke to Bruce about this and we agreed it was better than not having it at all. Bruce pointed out that cross-compilation is still a scenario (for crossgen2) and that it is likely that regressions/improvements with MSVC transfer to Clang/GCC as well.

Reply:
It is on my backlog to clean up how we lay this file out. I think when I do that I will sort the entries here and point out which ones are cross-compiled to make sure there is less confusion there. But let me leave that for that change.